src/prototype.rs

```diff
@@ -318,6 +429,26 @@ impl<'a> FirstPass<'a> {
         }
     }
 
+
+    /// Appends a line of HTML to the tree.
+    fn append_html_line(&mut self, start: usize, end: usize)
+    {
+
+        if end > start {
+            self.tree.append(Item {
+                start: start,
+                // Normalize CRLF to LF
+                end: if self.text.as_bytes()[end - 2] == b'\r' { end - 2 } else { end - 1},
```

Comment:
This looks like it will clip the last byte in the case where the line doesn't end in a newline.